### PR TITLE
Malloc and Free - a first pass

### PR DESCRIPTION
**Pull request by JMike360**
My very own trash implementation of malloc and free, please help

### CODE REVIEW

File: hal/jmos-stm32-periph.h
Comment (misaellopezg) [inline hal/jmos-stm32-periph.h:1]: I'm a n00b and don't remember this, did you literally define type unit32 to then define type reg_t?

Comment (JMike360) [reply hal/jmos-stm32-periph.h:1]: Yeah - uint32 is just a macro basically for "unsigned long int", and reg_t is *volatile* uint32

File: jmos/jmos-math.c
Comment (misaellopezg) [inline jmos/jmos-math.c:1]: don't forget to add comments for udiv and umod! Helps me understand it's unsigned division. udiv looks good!

Comment (JMike360) [reply jmos/jmos-math.c:1]: Added comments

**Pull request merged**